### PR TITLE
added adjustments for requests and boto3

### PR DESCRIPTION
**Pull request by SammyEK**
- fix a bug when the API response is not a json
- add a new logical naming strat (only for boto3, needs to be adapted for other modules)
- replace api parameters in file location and name

Signed-off-by: Sammy El Khammal <sammy.elkhammal@kensu.io>

### CODE REVIEW

File: kensu/utils/helpers.py [hunk 1]
```diff
@@ -74,4 +74,9 @@ def flatten(d, parent_key='', sep='.'):
                         items.extend(flatten(i, new_key, sep=sep).items())
             else:
                 items.append((new_key, type(v).__name__))
-    return dict(items)
\ No newline at end of file
+    return dict(items)
+
+def logical_naming_batch(string):
```
Comment (vidma) [inline kensu/utils/helpers.py:79]: weird name. what does it do? add a comment?

if it replaces all digits by # (e.g. `/api/user-id/123` -> `/api/user-id/###`) we could use a simpler regexp replace for that.

File: kensu/requests/api.py [hunk 2]
```diff
@@ -11,33 +13,49 @@ def wrapper(*args, **kwargs):
 
         kensu = KensuProvider().instance()
 
-        result_pk = DataSourcePK(location=result.url,
-                                 physical_location_ref=kensu.default_physical_location_ref)
-        result_ds = DataSource(name=result.url, format='API',
-                               pk=result_pk)._report()
+        try:
+            fields_dict = flatten(result.json())
```
Comment (vidma) [inline kensu/requests/api.py:17]: p.s. I think there should be a separate try/catch only for json. if json parsing fails, we use some arbitrary field name, like `data/value`?

File: kensu/requests/api.py [hunk 3]
```diff
@@ -11,33 +13,49 @@ def wrapper(*args, **kwargs):
...
 
+            # Construct the file location
 
-        fields_dict = flatten(result.json())
+            url = kwargs['url'] if 'url' in kwargs else args[0]
+            try:
+                params = kwargs['params'] if 'params' in kwargs else args[1]
```
Comment (vidma) [inline kensu/requests/api.py:23]: hmm, but what if `args[1]` is not params, e.g. no params passed at all...?
may also help to check type of `params` to be as expected - a `dict`

File: kensu/requests/api.py [hunk 4]
```diff
@@ -11,33 +13,49 @@ def wrapper(*args, **kwargs):
...
+            result_ds = DataSource(name=location, format='API',
+                                   pk=result_pk)._report()
+
+            fields = [FieldDef(name = k,  field_type=fields_dict[k], nullable=True ) for k in  fields_dict]
+            sc_pk = SchemaPK(result_ds.to_ref(),
+                             fields=fields)
+
+            result_sc = Schema(name="schema:" + result_ds.name, pk=sc_pk)._report()
+
+            result.__class__ = Response
```
Comment (vidma) [inline kensu/requests/api.py:44]: hmm, we do not modify the `result` class (just add monkey attributes), so this line is very likely not needed

File: kensu/requests/api.py [hunk 5]
```diff
@@ -11,33 +13,49 @@ def wrapper(*args, **kwargs):
...
+            result.ksu_schema = result_sc
+            result.ds_location = result.url
+            try:
+                import json
+                d = json.loads(result.text)
```
Comment (vidma) [inline kensu/requests/api.py:49]: we parsed the same json above one more time here https://github.com/kensuio-oss/kensu-py/pull/40/files#diff-af9d5001397882b10f679da56e5d65eb120620ddaa792f56d30ca1046399bb5cR17

**Review by vidma — APPROVED** [review]

File: kensu/utils/helpers.py (hunk 1 above)
Comment (SammyEK) [reply kensu/utils/helpers.py:79]: logical_naming_batch replace digits and group of digits with a simple #

File: kensu/requests/api.py (hunk 2 above)
Comment (SammyEK) [reply kensu/requests/api.py:17]: ok, foe now we only handle the case where result is a json/dict

File: kensu/requests/api.py (hunk 3 above)
Comment (SammyEK) [reply kensu/requests/api.py:23]: yep then `params = None` in the `except`

Comment (vidma) [reply kensu/requests/api.py:23]: i mean in theory we can get any other rubish in `args[1]`, isn't that possible ? unless we checked all usages of it

File: kensu/utils/helpers.py (hunk 1 above)
Comment (vidma) [reply kensu/utils/helpers.py:79]: so `/api/user-id/123 -> /api/user-id/#` ?

Comment (SammyEK) [reply kensu/utils/helpers.py:79]: exact

**Pull request merged**